src/mpid/ch4/src/mpidig_part.c

```diff
@@ -175,3 +170,56 @@ int MPIDIG_mpi_precv_init(void *buf, int partitions, MPI_Aint count,
   fn_fail:
     goto fn_exit;
 }
+
+int MPIDIG_part_start(MPIR_Request * request)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_ENTER;
+
+    //TG: why do we need to lock the VCI here?
+    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
+
+    /* activated request cannot be deleted unless a wait/test has been called */
+    MPIR_Part_request_activate(request);
+
+    /* No need to increase refcnt for comm and datatype objects,
+     * because it is erroneous to free an active partitioned req if it is not complete.*/
+    // TG: why not? I am really not clear on this reference counter management...
```

Comment:
the references to those objects are added at init time. there is no need to add additional references for each iteration. just release them when the request is freed.